src/gh-api-client.ts

```diff
@@ -0,0 +1,114 @@
+import * as httpm from "@actions/http-client";
+
+const userAgent = "GitHub multi-repository variant analysis action";
+
+export function getApiClient() {
+  return new httpm.HttpClient(userAgent, [], {
+    allowRetries: true,
+  });
+}
+
+interface InProgressProperties {
```

Comment:
These names have room for improvement. We're in the context of a gh-api-client so "InProgressProperties" sounds a bit too generic to me.

How about `InProgressAnalysis`, `SuccessfulAnalysis` etc.?